mungegithub/mungers/approvers/owners.go

```diff
@@ -137,15 +137,46 @@ func findMostCoveringApprover(allApprovers []string, reverseMap map[string]sets.
 	return bestPerson
 }
 
+// temporaryUnapprovedFiles returns the list of files that wouldn't be
+// approved by the given set of approvers.
+func (o Owners) temporaryUnapprovedFiles(approvers sets.String) sets.String {
+	ap := NewApprovers(o)
+	for approver := range approvers {
+		ap.AddApprover(approver, "", "")
+	}
+	return ap.UnapprovedFiles()
+}
+
+// KeepMostCoveringApprovers finds who we should keep as suggested approvers given a pre-selection
+// knownApprovers must be a subset of potentialApprovers.
+func (o Owners) KeepMostCoveringApprovers(knownApprovers sets.String, potentialApprovers []string) sets.String {
+	keptApprovers := sets.NewString()
+
+	unapproved := o.temporaryUnapprovedFiles(knownApprovers)
+	reverseMap := o.GetReverseMap()
+
+	for _, suggestedApprover := range o.GetSuggestedApprovers(potentialApprovers).List() {
+		if reverseMap[suggestedApprover].Intersection(unapproved).Len() != 0 {
```

Comment:
> Yes. Can you generate a name then? :-)
Since there's no max, how about "KeepCoveringApprovers"

Ack on your other comments.  I think we need to make a PR to assign more uniformly using the reviewers pool now.